lnwallet/wallet.go

```diff
@@ -1380,6 +1417,119 @@ func (l *LightningWallet) selectCoinsAndChange(feeRate SatPerKWeight,
 	return inputs, changeOutputs, unlock, nil
 }
 
+// selectCoinsToSpend performs coin selection in order to spend exactly
+// 'toSpend' creating the funding tx. If necessary, a change address will also
+// be generated. If coin selection is successful/possible, then the selected
+// coins and change outputs are returned, and the value of the resulting
+// funding output. This method locks the selected outputs, and a function
+// closure to unlock them in case of an error is returned.
+//
+// NOTE: This method is different from selectCoinsAndChange in the way that it
+// spends the given amount when selecting coins, instead of targetting a
+// specific funding output size.
+func (l *LightningWallet) selectCoinsToSpend(feeRate SatPerKWeight,
```

Comment:
There's a good bit of duplication here with `coinSelect` and `selectCoinsAndChange`. Seems like it'd be possible to refactor some things around.

Reply:
Done.

Reply:
Hmm I was thinking of actually reusing the existing `selectCoins` method. We'd pass a `subtractFees bool` to it and, if set, run the special case logic.

Reply:
Done.